src/Adapter/Module/AdminModuleDataProvider.php

```diff
@@ -177,97 +187,73 @@ public function isAllowedAccess($action, $name = '')
     }
 
     /**
-     * @param AddonsCollection $addons
+     * @param ModuleCollection $modules
      * @param string|null $specific_action
      *
-     * @return AddonsCollection
+     * @return ModuleCollection
      */
-    public function generateAddonsUrls(AddonsCollection $addons, $specific_action = null)
+    public function setActionUrls(ModuleCollection $modules, ?string $specific_action = null): ModuleCollection
     {
-        foreach ($addons as $addon) {
+        foreach ($modules as $module) {
```

Comment:
Since module attributes are frequently accessed in this method, I suggest an optimization with a new dedicated getter method right here:
```
        $moduleAttributes = $module->getAttributes();
```